ingest/lib/index-rectifier.js

```diff
@@ -0,0 +1,186 @@
+// @ts-check
+const pMap = require('p-map');
+
+const dspaceClient = require('../../lib/dspace-client');
+const osClient = require('../../lib/open-search-client');
+const utils = require('./ingest-queue');
+
+module.exports = {
+  /**
+   * Commits index items that have been marked as out of date by
+   * queuing them for re-ingest. Queuing will be done in parallel.
+   *
+   * @param {Object[]} items List of existing index items that need to
+   *                         be updated.
+   * @param {string} ingestTopicArn SNS Topic ARN where new documents
+   *                                are queued.
+   * @param {Object} [options={}] Additional options.
+   * @param {string} [options.region=us-east-1] AWS region containing the
+   *                                            infrastructure.
+   */
+  commitUpdatedItems: async (
+    items,
+    ingestTopicArn,
+    options = {}
+  ) => {
+    if (items.length <= 0) return;
+
+    const { region = 'us-east-1' } = options;
+
+    await pMap(
+      items,
+      async (item) => {
+        try {
+          await utils.queueIngestDocument(
+            // eslint-disable-next-line no-underscore-dangle
+            item._id,
+            ingestTopicArn,
+            { region }
+          );
+        } catch (error) {
+          console.log(`ERROR: Failed to queue updated document ${JSON.stringify(item)} with error: ${error}`);
+        }
+      },
+      { concurrency: 5 }
+    );
+  },
+
+  /**
+   * Commits removed items that have been marked as removed by
+   * deleting them from the index.
+   *
+   * @param {Object[]} items List of existing index items that need to
```

Comment:
`Object[]` is treated as `any[]`, which isn't real helpful. Can we make this more specific?